### DISCUSSION TITLE
[HELP] En un campo de texto, cómo puedo separar correos?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#244] [2. Questions and Answers] [answered]
Sé que no se puede, o quizas si, pero un campo de texto, donde escriba nicolas.pena@cotalker.com y administrador@cotalker.com,  el sistema me indique que son 2 correos para enviar en forma de copia?

Como no se puede hacer eso, creo, tengo 2 soluciones:

1.- Agregar 3 campos de textos para cada correo
2.- Agregar un survey/survey de texto para que me ingrese el correo.

Aunque ambas soluciones me convergen a una sola pregunta:

Cómo los puedo dejar como en el campo destinatario los correos que ingreso?. No puedo dejarlos unitarios porque el que se encuentre en copia le llegará N correos, donde N es la cantidad de correos que indico.

Se puede hacer esto:

1.- para los 3 campos, los puedo colocar por separados, en el correo, tendría problema si no me escriben 1  de esos 3 campos, ya que el sistema no encontraría 1 y por ende no me enviaría el correo

2.- Para el survey/survey, podría utilizar un PBIterate para el campo del s/s, pero tendría problema en agregar los correos de texto al PBemail.

Gracias por su ayuda :>

**Comment by guillevalin** [comment, marked as answer]
@NicolasPLG Podrías usar un solo campo de texto, y que te ingresen los correos con un separador (ej: punto y coma).

nicolas@cotalker.com;julio@cotalker.com

Usando la rutina Custom JavaScript podrías hacer un split de ese valor por el separador (;) y luego pasarselo a otra etapa como valores independientes $OUTPUT#split_emails|0 por ejemplo.

**Reply by NicolasPLG** [reply]
pero tendría que usar antes un iterate o algo así. o colocar los campos: $OUTPUT#split_emails|0, $OUTPUT#split_emails|1, $OUTPUT#split_emails|2 esperando que no haya problemas cuando tenga 2 correos y no 3? o sería correos por separados?

**Reply by guillevalin** [reply]
Pero lo que no te entiendo, es que quieres enviar emails por separado o un solo email?

**Reply by NicolasPLG** [reply]
Tiene que ser solo en un correo, para no duplicar los correos al que va en copia.
Serviría si agrego eso en un schema node en el asset de nombre: "correo", y que en el target colocar $output#nwr_assets#data|schemanodes|correo? (obviamente la escritura está mal, pero es más que nada en la idea).

**Reply by NicolasPLG** [reply]
Encontré un cjs que debería funcionar similar pero no entiendo bien el src:


```
{
                "_id" : ObjectId("6099715e3e26400009780dbc"), 
                "key" : "custom_code", 
                "name" : "CCJS", 
                "version" : null, 
                "data" : {
                    "src" : "```const x = { } ;   value.email.split('####').forEach((a, b) => {  x['line_' + b] = a;  });   return x; ```"
                }, 
                "next" : {
                    "DEFAULT" : "message"
                }, 
                "customNetworkRequest" : [

                ]
            }, 
            {
                "_id" : ObjectId("6099743c2f0eac0008eb0454"), 
                "key" : "message", 
                "name" : "PBMessage", 
                "version" : null, 
                "data" : {
                    "content" : "$JOIN##```__Nueva Alerta__ (Email):  __Nombre:__```#($OUTPUT#custom_code#data|line_1)#```__Sensor:__```#($OUTPUT#custom_code#data|line_2)#```__Alerta__```#($OUTPUT#custom_code#data|line_3)#($OUTPUT#custom_code#data|line_4)#($OUTPUT#custom_code#data|line_5)", 
                    "contentType" : "text/system", 
                    "sentBy" : "607265d3c124247b8c24472a", 
                    "channelIds" : [
                        "607265dac1242405a6244a93"
                    ]
                }, 
                "next" : {
                    "DEFAULT" : ""
                }, 
                "customNetworkRequest" : [

                ]
            }
```

Lo que debería hacer en el custom seria algo asi (creo):

```
   "data" : {
                    "src" : "```const x = { } ;   value.data.split(value.data.find(x => x.identifier === 'question'),';').forEach((a, b) => {  x['line_' + b] = a;  });   return x; ```"
                }, 
```

No sé si el con lo que encontré, cabmio sólo el "####" por ";" me debería funcionar.

:(

**Reply by guillevalin** [reply]
Ese Custom JS lo que hace es splitear un string separado por ####. Si los correos los tienes separados por ";", puedes usar ";" en el split.

Mi duda es como quieres usar el arreglo de correos. Según el ejemplo te quedarán en un arreglo con formato:
$OUTPUT#custom_code#data|line_0  ... line_1 ... line_2 ... etc

**Reply by NicolasPLG** [reply]
Quiero que quedase asi

![image](https://user-images.githubusercontent.com/61419471/121247855-b72cf000-c870-11eb-8c56-279fc833c8a5.png)

Entonces estará bien reutilizar ese código tal cual como está (cambiando los #### por ; y los pueda separar de esa forma o debo agregar algo mas?. lo pregunto porque no sé como quedaría si está asociado a una question

**Reply by guillevalin** [reply]
Está correcto así @NicolasPLG

**Reply by NicolasPLG** [reply]
Perfecto gracias @guillevalin reutilizaré el código para dejarlo escrito

Cerraré el tema
